src/main/java/mekanism/common/tile/prefab/TileEntityStructuralMultiblock.java

```diff
@@ -14,25 +12,27 @@
 import mekanism.common.lib.multiblock.MultiblockManager;
 import mekanism.common.lib.multiblock.Structure;
 import mekanism.common.tile.base.TileEntityMekanism;
-import mekanism.common.util.MekanismUtils;
 import net.minecraft.core.BlockPos;
 import net.minecraft.nbt.CompoundTag;
-import net.minecraft.nbt.Tag;
 import net.minecraft.world.InteractionHand;
 import net.minecraft.world.InteractionResult;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.state.BlockState;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 public abstract class TileEntityStructuralMultiblock extends TileEntityMekanism implements IStructuralMultiblock, IConfigurable {
 
-    private final Map<MultiblockManager<?>, Structure> structures = new HashMap<>();
+    //Note: We never expect this to actually be filled, but we set the default to six in case it somehow is part of a structure in every direction
+    // given our expected is so low we use an array map
+    private final Map<MultiblockManager<?>, Structure> structures = new Reference2ObjectArrayMap<>(6);
```

Comment:
Won't this break down if the block is part of two of the same type? e.g. 2 boilers
managers are singletons as far as I can see

the capacity should be smaller though, I'd go with 2 at most

Reply:
Does three make more sense given the fact it contains `null` and an actual formed structure?

Reply:
Also the way that multiblock formation currently works is if there are two frames that are remotely close to touching of the same type, then they get combined into a single multiblock and won't form/ So it can't be part of two of the same type. (granted there aren't cases we can't have it be part of two multiblocks currently regardless)

Reply:
You would have to ask aidan. I believe it is for when structural multiblocks are connected together but are not part of a multiblock type yet. I semi tried to remove it as part of this PR and it was causing some weird edge cases where multiblock contents were getting voided...

Reply:
@pupnewfster please don't use ArrayMaps from fastUtil.
they are simply arraylist implementations that do a simply indexOf search for finding the key.
They are horrible in performance and access times while the iteration is faster technically, it is not that much faster then a linkedMap which is extremely similar to the datastructure like arrayMaps are.

Reply:
First off, we don't really care about the iteration order, but secondly for tiny maps array maps do have better performance than linked maps.